tests/src/3d/testqgsaabb.cpp

```diff
@@ -43,6 +43,7 @@ void TestQgsAABB::initTestCase()
 //runs after all tests
 void TestQgsAABB::cleanupTestCase()
 {
+  QgsApplication::exitQgis();
```

Comment:
I feel a bit uneasy if there's exitQgis() but no initQgis() in the initTestCase() metod :grin:  - but I guess that's intentional, to clean up everything, even if we did not need to fully initialize the environment, right?

Reply:
Actually the issue is that QGSTEST_MAIN creates the QgsApplication -- so even tests which don't explicitly call initQgis need to exitQgis in order to cleanly exit. It's obviously not strictly necessary, but avoids noise when running under valgrind/asan....

Reply:
I see... thanks!

I just hope one day we will refactor QgsApplication - there are quite a few confusing bits :grinning: